lib/logger/lib/logger/backends/console.ex

```diff
@@ -97,4 +163,52 @@ defmodule Logger.Backends.Console do
   defp color_event(data, level, %{enabled: true} = colors) do
     [IO.ANSI.format_fragment(Map.fetch!(colors, level), true), data | IO.ANSI.reset]
   end
+
+  defp log_buffer(%{buffer_size: 0, buffer: []} = state), do: state
+
+  defp log_buffer(state) do
+    %{device: device, buffer: buffer, buffer_ts: buffer_ts} = state
+    %{state | ref: async_io(device, buffer), buffer: [], buffer_size: 0,
+      buffer_ts: nil, last_ts: buffer_ts}
+  end
+
+  defp handle_error(error, %{last_ts: :error}) do
```

Comment:
If last_ts is :error it means we failed to put the buffer/event and failed
to put the retry with known Unicode data. Previously an ArgumentError would
be raised in this situation.

On Tuesday, 31 May 2016, José Valim <notifications@github.com> wrote:

> In lib/logger/lib/logger/backends/console.ex
> <https://github.com/elixir-lang/elixir/pull/4728#discussion_r65135076>:
>
> > @@ -97,4 +163,52 @@ defmodule Logger.Backends.Console do
> >    defp color_event(data, level, %{enabled: true} = colors) do
> >      [IO.ANSI.format_fragment(Map.fetch!(colors, level), true), data | IO.ANSI.reset]
> >    end
> > +
> > +  defp log_buffer(%{buffer_size: 0, buffer: []} = state), do: state
> > +
> > +  defp log_buffer(state) do
> > +    %{device: device, buffer: buffer, buffer_ts: buffer_ts} = state
> > +    %{state | ref: async_io(device, buffer), buffer: [], buffer_size: 0,
> > +      buffer_ts: nil, last_ts: buffer_ts}
> > +  end
> > +
> > +  defp handle_error(error, %{last_ts: :error}) do
>
> Why are we raising in those cases? Can we get some docs? It was not
> obvious from looking at the code.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/elixir-lang/elixir/pull/4728/files/8ea89cbab4b6187a0e7f3de023171d1203a33a90#r65135076>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AB6JTdmzrgRsvMYCtYbXc0FpJ10INoWUks5qG-XWgaJpZM4Ip-Xy>
> .
>